### PR TITLE
Use backend-types properly, and fix minor lint errors

### PR DESCRIPTION
**Pull request by jonasbjoralt**
- Make read/write models have non-nullable types
- Refrence them directly frontend
- Fix Abstences typo
- Fix minor type errors

### CODE REVIEW

File: frontend/src/components/FilteredConsultantsList.tsx [hunk 1]
```diff
@@ -33,7 +34,7 @@ export default function StaffingTable() {
         <col span={1} className="w-[190px]" />
         {filteredConsultants
           .at(0)
-          ?.bookings?.map((booking, index) => <col key={index} span={1} />)}
+          ?.bookings.map((x, index: number) => <col key={index} span={1} />)}
```
Comment (yelodevopsi) [inline frontend/src/components/FilteredConsultantsList.tsx:37]: use `_unused` or some other _underscore varname instead?

File: frontend/src/types.ts
Comment (yelodevopsi) [inline frontend/src/types.ts:1]: Dont we have a lot of custom types here? Where are these placed?

**Review by yelodevopsi — DISMISSED** [review]
Really good work! 👏  🎉  Appreciate the usage of the types generated from swagger. 

Edit: Watch the comments I want resolved/discussed.

File: frontend/src/types.ts
Comment (jonasbjoralt) [reply frontend/src/types.ts:1]: Nope, all types except the two that remain were from backend

File: frontend/src/components/Staffing/DetailedBookingRows.tsx [hunk 2]
```diff
@@ -87,29 +87,30 @@ interface updateBookingHoursProps {
...
     hours: props.hours,
-    bookingType: props.bookingType,
+    type: props.bookingType,
     consultantId: props.consultantId,
-    bookingId: props.bookingId,
+    engagementId: props.bookingId,
     startWeek: props.startWeek,
-    endWeek: props.endWeek,
+    startYear: 2023, // TODO: Fix mismatch between front- and backend types
+    //endWeek: props.endWeek,
```
Comment (sigtheidiot) [inline frontend/src/components/Staffing/DetailedBookingRows.tsx:105]: This will maybe leave the drag function useless? as it needs an end week

Comment (sigtheidiot) [reply frontend/src/components/Staffing/DetailedBookingRows.tsx:105]: The reason for a different type here and in the route handler was to enable the use of a singular route handler for two different api end points with different types.

**Review by sigtheidiot — APPROVED** [review]

**Pull request merged**